### PR TITLE
⚓️ Dynamic Proxy Port

### PR DESCRIPTION
**Pull request by Gilad-Gur-Andelman**

### CODE REVIEW

**Comment by idoco** [comment]
Can you share a video/screenshot of the customization process?

File: src/renderer-process/proxy-window/ui/proxy-dashboard.tsx [hunk 1]
```diff
@@ -55,6 +57,7 @@ export const ProxyDashboard = (): JSX.Element => {
     window.desktopApi.getIpAddress();
     window.desktopApi.initFilterRegex();
     window.desktopApi.isRecording();
+    window.desktopApi.port();
```
Comment (onniaa) [inline src/renderer-process/proxy-window/ui/proxy-dashboard.tsx:60]: what does this do?

Comment (Gilad-Gur-Andelman) [reply src/renderer-process/proxy-window/ui/proxy-dashboard.tsx:60]: Good question. It gets the used port in proxy server (recorder). Maybe it needs a rename.

File: src/main-process/proxy/port.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import GetPort from 'get-port';
```
[the remark this answers is not in the document]
Comment (Gilad-Gur-Andelman) [reply src/main-process/proxy/port.ts:1]: Yes, I had a local function named `getPort` I didn't want them to clash but then I deleted my function and forgot to rename this import.

**Comment by yigaldviri** [comment]
This is a test comment

**Comment by idoco** [comment]
> > Can you share a video/screenshot of the customization process?
> 
> From the [Trello ticket](https://trello.com/c/EkgVGcbH/2144-desktop-app-dynamic-port-number-for-proxy-recorder):
> 
> **The current 1234 port number is not not good, because it is a constant (could be occupied) and it is not a conventional port number. Will change to a dynamic open port, preferring 8080, 8081, 9090, 9091.**
> 
> Also, if the above ports are not available, we will pick a random open port (using [get-port](https://www.npmjs.com/package/get-port) npm package)

I'm not sure that port searching will be good enough. What if you start your backend uses that port but you only start it after you start the recorder?

Wouldn’t it be simpler to go for a default value and allow the user to choose a different one?

**Comment by Gilad-Gur-Andelman** [comment]
> What if your backend uses that port, but you only start it after you start the recorder?

I don't understand the question. The proxy server starts listening on the port as soon as the app is up and running.

> Wouldn’t it be simpler to go for a default value and allow the user to choose a different one?

A default value is not safe since this value might not be available.
To "allow the user to choose a different one" we need to define how this will be done.

**Comment by Gilad-Gur-Andelman** [comment]
Artifacts are available here: https://output.circle-artifacts.com/output/job/900f9a85-ec41-40eb-9cd7-4d636c3e6f06/artifacts/0/out/make/Loadmill-3.0.1-x64.dmg
https://output.circle-artifacts.com/output/job/900f9a85-ec41-40eb-9cd7-4d636c3e6f06/artifacts/0/out/make/zip/darwin/x64/Loadmill-darwin-x64-3.0.1.zip

**Comment by Gilad-Gur-Andelman** [comment]
Artifacts are available here:
https://output.circle-artifacts.com/output/job/900f9a85-ec41-40eb-9cd7-4d636c3e6f06/artifacts/0/out/make/Loadmill-3.0.1-x64.dmg

https://output.circle-artifacts.com/output/job/900f9a85-ec41-40eb-9cd7-4d636c3e6f06/artifacts/0/out/make/zip/darwin/x64/Loadmill-darwin-x64-3.0.1.zip

**Comment by Gilad-Gur-Andelman** [comment]
## 🏗 Build Artifacts are available here👇 

### 1. [🍏 .dmg](https://output.circle-artifacts.com/output/job/90190d39-6fd2-46e6-86c8-9b88c2efdeaf/artifacts/0/out/make/Loadmill-3.0.1-x64.dmg)

### 2. [🤐 .zip](https://output.circle-artifacts.com/output/job/90190d39-6fd2-46e6-86c8-9b88c2efdeaf/artifacts/0/out/make/zip/darwin/x64/Loadmill-darwin-x64-3.0.1.zip)

**Review by yigaldviri — APPROVED** [review]

**Pull request merged**